### PR TITLE
Mock DbErr

### LINKED ISSUE #1235: Support configuring `MockDatabase` to return `DbError` on all types of operations against it.
## Motivation

With the addition of fine-grained database errors added in `0.10` release, we've written some application code to respond to different errors intelligently.  We'd like to unit test this code.  Currently we have found no way to configure `MockDatabase` to return a `DbErr` for a query result or an execution result.

## Proposed Solutions

Wrap the input to `append_exec_results` and `append_query_results` to accept Result variants instead of just raw values?  

```rust
    /// Add the [MockExecResult]s to the `exec_results` field for `Self`
    pub fn append_exec_results(mut self, mut vec: Vec<MockExecResult>) -> Self {
        self.exec_results.append(&mut vec);
        self
    }

    /// Add the [MockExecResult]s to the `exec_results` field for `Self`
    pub fn append_query_results<T>(mut self, vec: Vec<Vec<T>>) -> Self
    where
        T: IntoMockRow,
    {
        for row in vec.into_iter() {
            let row = row.into_iter().map(|vec| vec.into_mock_row()).collect();
            self.query_results.push(row);
        }
        self
    }
```

### PR DESCRIPTION
**Pull request by tyt2y3**
Resolves #1235

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -116,4 +116,4 @@ runtime-tokio-rustls = [
     "sea-query-binder?/runtime-tokio-rustls",
     "runtime-tokio",
 ]
-tests-cfg = []
+tests-cfg = ["serde/derive"]
```
Comment (tyt2y3) [inline Cargo.toml:119]: This is needed for 

```
cargo t --lib --features=tests-cfg,with-json -- mock
```

**Comment by tyt2y3** [comment]
@robot-head does this work for you?

**Comment by robot-head** [comment]
Looks good to me! Great job, thank you

**Review by robot-head — APPROVED** [review]

**Review by billy1624 — APPROVED** [review]
What a smart way to avoid breaking existing public API :)

**Comment by robot-head** [comment]
> What a smart way to avoid breaking existing public API :)

Yeah honestly impressed, a much better solution than I proposed, learning here!

**Comment by tyt2y3** [comment]
Feeling flattered.

https://github.com/SeaQL/sea-orm/releases/tag/0.10.4

**Pull request merged**